nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementError
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0..10]
```

Comment:
- [ ] The maximum characters for the name should be 10, so remember you always start counting from 0, so right now it is taking in 11 characters in total, so please work on this issue.
- [ ] Also, it would be nice to make some kind of condition so the code does something if the name is over 10 characters, so you could do something like:
```
def correct_name
  @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.corect_name
end
```
This way, you are trimming the name if it is longer than 10 characters, otherwise you execute the correct method.

Reply:
Thank you for your review. I have worked on that.